### ISSUE TITLE
Module : Community maintained Machine/Part Catalog

### ISSUE THREAD

**Issue by tx4x** [OneArmyWorld/onearmy#274]
Hi,

I can't find anything about how the machines are handled, there was some mentioning of doing the bazar from scratch but no idea this was real. 

However, to me, all the modules in progress (and the planned ones i could find) are pretty much secondary in comparison to an extendable machine/part catalog. Most folks get active with getting machines and/or parts first and rarely I met someone of our users/clients who got active in the forum, before or after. 

Currently there is already quit a disaster or mess: the download kit's material bill is partly done beyond any reality, and worst, often extreme misleading and incomplete, rather of damaging nature and in the bazar on the other hand there is mostly a hefty labor price tag on all, creating even more confusion. So that are quite different and competing realities distorting the possibilities/opportunities for new users (which i am tired off to correct many times the week). 

I also have no idea whether v3 gets abandoned or not. There is so much valuable feedback and updates in the forum that v3 should get an update first. There is enough for a v3.4 at least.

However, @mattia-io , could you please bring some light here. I am willing to get this done, ASAP, an if it last til December, I don't mind. The how-to section is cool and seems like a more simple clone of all the other DIY services/platform but still nothing really useful compared to a community maintained machine catalog. The discussion module, well. It's just there because it has to, and it's going to be a messy place :-) Anyways, if you feel uncomfortable with the task regarding data-structure, implementation details, ux and whatever, I can do a draft design of what's needed, also providing the means to build the new bazar on top. But long story short, here is what comes to mind: 

- convert v3 to machine/parts, migrate v3 user feedback. i've done quite some stuff in this direction in our infrastructure already.
- enable machine versioning, similar to 'instructables' so user can up-vote custom builds, thus new ideas get more visibility and eventually get approved as machine update, by the community
- machine builds with enough scores can be considered as valuable update
- each machine ideally has references to sources, services or supplies
- so basically the idea is also to decentralize PP and run it as network :-) so you guys can enjoy more off-time with that as well :-) 
- interface vise this could get really cute, an editable image-map could have links to alternative components, user feedback, etc...

needless to say, I also understand such module to enable machine builders to actually start working together, secure and share suppliers, etc...... you know: that one-army thing :-) Because, right now it's pretty poor to me, most folk copy & paste v3 with apparently not much thinking and sell it afterwards for big $$ with pretty much no innovation or return to the network.... doesn't need to be like this :-)

**Comment by tx4x** [comment]
or how else could users add complete data for machines like : press, vacuum table, blow table, etc... ?
